### DISCUSSION TITLE
Welcome to obsidian-regex-pipeline Discussions!

### DISCUSSION THREAD

**Discussion by No3371** [No3371/obsidian-regex-pipeline#14] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * **Share rulesets in the Rulesets category.**
  * Tell us what do you think about the plugin.

**Comment by precurcor** [comment]
Is there anyway to replicate the list matches feature that is on [regexr](https://regexr.com/)
Really just removing every thing but the matches?

**Reply by No3371** [reply]
Should be achievable with Backreference, which is a feature of Regex.
```
ABCDEF
```
will be replaced into
```
B
D
F
```
with the rule:
```
"A(.)C(.)E(.)"->"$1
$2
$3"
```

**Reply by J1Pestes** [reply]
**Ruleset**
Does this do what you want?
```
".*?(term).*?"->"$1
"
"(?:(?!term).)*"s->" "x
"\s*^.*$(?!\s)"->" "x
```
**Example with term replaced with complexity:**

_Original_
Instead of thinking about time, think about the number of elementary operations performed as a function of the size of the input (i.e. the activity to be performed).
This idea is inspired by the concept of [time complexity](https://en.wikipedia.org/wiki/Special:Search?search=time%20complexity) from [computational complexity theory](https://en.wikipedia.org/wiki/Special:Search?search=computational%20complexity%20theory). I find other kinds of [computational complexity](https://en.wikipedia.org/wiki/Special:Search?search=computational%20complexity) inspirational for personal productivity too (e.g. space ~ and communication ~ (for group projects)). I came up with the idea of generalising computational complexity for personal complexixty when I saw [this table in an awesome YouTube video](https://youtu.be/AAwYzYkjNTg?t=863).
_Replacement_
complexity
complexity
complexity
complexity
complexity
complexity
complexity

**Notes**
The regex searches inside hidden text too. For example, it retrieved complexity from `https://en.wikipedia.org/wiki/Special:Search?search=time complexity` which is hidden in obsidian using regular Markdown link syntax (`[]()`).
